arcgis-ios-sdk-samples/Edit data/Create and save KML file/CreateAndSaveKMLViewController.swift

```diff
@@ -0,0 +1,286 @@
+        let kmzProvider = KMZProvider(document: kmlDocument)
+        let activityViewController = UIActivityViewController(activityItems: [kmzProvider], applicationActivities: nil)
+        activityViewController.popoverPresentationController?.barButtonItem = sender
+        present(activityViewController, animated: true)
+        activityViewController.completionWithItemsHandler = { (activityType: UIActivity.ActivityType?, completed: Bool, arrayReturnedItems: [Any]?, error: Error?) in
+            kmzProvider.deleteKMZ()
+        }
+    }
+
+    // Complete the current sketch and add it to the KML document.
+    @IBAction func completeSketch() {
+        let geometry = mapView.sketchEditor?.geometry
+        let projectedGeometry = AGSGeometryEngine.projectGeometry(geometry!, to: spatialRef)
+        let kmlGeometry = AGSKMLGeometry(geometry: projectedGeometry!, altitudeMode: .clampToGround)
+        currentPlacemark = AGSKMLPlacemark(geometry: kmlGeometry!)
+        currentPlacemark!.style = kmlStyle
+        kmlDocument.addChildNode(currentPlacemark!)
+        mapView.sketchEditor?.stop()
+        updateToolbarItems()
+        actionButtonItem?.isEnabled = true
+    }
+    
+    // Reset the KML.
+    @IBAction func resetKML() {
+        mapView.map?.operationalLayers.removeAllObjects()
+        currentPlacemark = nil
+        kmlDocument = AGSKMLDocument()
+        let kmlDataset = AGSKMLDataset(rootNode: kmlDocument)
+        kmlLayer = AGSKMLLayer(kmlDataset: kmlDataset)
+        mapView.map?.operationalLayers.add(kmlLayer!)
+    }
+    
+    var sketchCreationMode: AGSSketchCreationMode?
+    var kmlDocument = AGSKMLDocument()
+    let spatialRef = AGSSpatialReference.wgs84()
+    var kmlStyle = AGSKMLStyle()
+    var currentPlacemark: AGSKMLPlacemark?
+    var kmlLayer: AGSKMLLayer?
+    
+    // Prompt icon selection action sheet.
+    func addPoint() {
+        let alertController = UIAlertController(title: "Select Icon", message: "This icon will be used for the new feature", preferredStyle: .actionSheet)
+        let icons: [(String, URL)] = [
+            ("No style", URL(string: "http://resources.esri.com/help/900/arcgisexplorer/sdk/doc/bitmaps/148cca9a-87a8-42bd-9da4-5fe427b6fb7b127.png")!),
+            ("Star", URL(string: "https://static.arcgis.com/images/Symbols/Shapes/BlueStarLargeB.png")!),
+            ("Diamond", URL(string: "https://static.arcgis.com/images/Symbols/Shapes/BlueDiamondLargeB.png")!),
+            ("Circle", URL(string: "https://static.arcgis.com/images/Symbols/Shapes/BlueCircleLargeB.png")!),
+            ("Square", URL(string: "https://static.arcgis.com/images/Symbols/Shapes/BlueSquareLargeB.png")!),
+            ("Round pin", URL(string: "https://static.arcgis.com/images/Symbols/Shapes/BluePin1LargeB.png")!),
+            ("Square pin", URL(string: "https://static.arcgis.com/images/Symbols/Shapes/BluePin2LargeB.png")!)
+        ]
+        icons.forEach { (title, url) in
```

Comment:
I recall one thing, but correct me if I'm wrong. The dictionary in Swift is an unordered map, which means the order of `icons` varies each time. Therefore it might be a good idea to sort it before mapping each choice to its action. 💭

Reply:
Or use an array of tuples if you don't really need a dict.

Reply:
> Or use an array of tuples

That is precisely what this is. The type of `icons` is `[(String, URL)]`. A dictionary would be `[String: URL]`.

Reply:
@vquach2404 However, `colors` in both `addPolyline()` and `addPolygon()` is a dictionary. Please use an array of tuples, like `icons`.

Reply:
Ohh just realized I left my comment on the wrong line... 😅 Phil Thanks for pointing to the right place!